lib/types.ts

```diff
@@ -119,7 +119,7 @@ class OrType<T> extends BaseType<T> {
   }
 }
 
-class PredicateType<T> extends BaseType<T> {
+export class PredicateType<T> extends BaseType<T> {
```

Comment:
Instead, we could change the return type of `defBuiltInType` to `Type<T>` instead of `PredicateType<T>` to hide the implementation detail that the built-in types are `PredicateType`s internally. I'll go ahead and make that change.